aten/src/ATen/Utils.h

```diff
@@ -93,10 +93,16 @@ inline int64_t sum_intlist(ArrayRef<int64_t> list) {
   return std::accumulate(list.begin(), list.end(), 0ll);
 }
 
-inline int64_t prod_intlist(ArrayRef<int64_t> list) {
-  return std::accumulate(list.begin(), list.end(), 1ll, std::multiplies<int64_t>());
+template<typename C, typename std::enable_if<std::is_integral<typename C::value_type>::value, int>::type = 0>
+inline int64_t prod_intlist(const C &container){
+    return std::accumulate(container.begin(), container.end(), static_cast<int64_t>(1), std::multiplies<int64_t>());
```

Comment:
Could you leave a comment here explaining the rationale for why this is setup as it is?

Reply:
I, at least, didn't know that `std::accumulate` behaved the way it does, and that seems like important knowledge to retain so the alias function isn't inadvertently simplified in the future.